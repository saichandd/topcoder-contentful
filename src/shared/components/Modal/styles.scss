@import '~styles/mixins';

$button-space-32: 6 * $base-unit;

:global {
  body.scrolling-disabled-by-modal {
    overflow: hidden;
  }
}

.overlay {
  background: $tc-gray-neutral-dark;
  border: none;
  height: 100%;
  left: 0;
  opacity: 0.8;
  outline: none;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 998;
}

.container {
  background: #fff;
  box-shadow: 0 0 14px 1px rgba(38, 38, 40, 0.15);
  border-radius: 2 * $corner-radius;
  max-height: 95vh;
  max-width: $screen-md;
  overflow: hidden;
  padding: 8 * $base-unit;
  width: 480px;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 999;

  @include xs-to-lg {
    max-width: 95vw;
  }

  @include xs-to-sm {
    padding: 40px 10px;
  }
}
