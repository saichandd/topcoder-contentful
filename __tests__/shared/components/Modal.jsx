import Modal from 'components/Modal';
import React from 'react';
import Rnd from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';

test('Snapshot match', () => {
  const rnd = new Rnd();
  rnd.render(<Modal />);
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});

class ModalClass extends React.Component {
  componentDidMount() {}

  render() {
    return <Modal {...this.props} />;
  }
}

const mockOnCancel = jest.fn();

const render = TU.renderIntoDocument((
  <ModalClass
    onCancel={mockOnCancel}
  />
));

test('onCancel', () => {
  const obj = TU.findAllInRenderedTree(render, item => item && item.className && item.className.match('overlay'))[0];
  TU.Simulate.click(obj);
  expect(mockOnCancel).toHaveBeenCalled();
});

test('onWheel', () => {
  const container = TU.findAllInRenderedTree(render, item => item && item.className && item.className.match('container'))[0];
  TU.Simulate.wheel(container);
  const overlay = TU.findAllInRenderedTree(render, item => item && item.className && item.className.match('overlay'))[0];
  TU.Simulate.wheel(overlay);
});
