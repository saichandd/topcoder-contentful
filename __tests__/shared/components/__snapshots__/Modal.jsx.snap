// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot match 1`] = `
<Modal
  onCancel={[Function]}
  theme={
    Object {
      "container": "src-shared-components-Modal-styles___container___2HN2GN",
      "overlay": "src-shared-components-Modal-styles___overlay___36ZAXn",
      "scrolling-disabled-by-modal": "src-shared-components-Modal-styles___scrolling-disabled-by-modal___1YgZ3T",
    }
  }
/>
`;
